python/sqlflow_submitter/xgboost/feature_column.py

```diff
@@ -170,16 +202,20 @@ def _set_field_names(self, field_names):
     def get_column_names(self):
         return [self.key]
 
+    def num_classes(self):
+        return self.hash_bucket_size
+
     def __call__(self, inputs):
         def transform_fn(slot_value):
             if isinstance(slot_value, np.ndarray):
-                output = np.ndarray(slot_value.shape)
+                output = np.ndarray(slot_value.shape, dtype=np.int64)
                 for i in six.moves.range(slot_value.size):
-                    output[i] = hashing(slot_value[i])
+                    output.put(
```

Comment:
Is `output[i] = hashing(slot_value.take(i)) % self.hash_bucket_size` cleaner?

Reply:
It may not work. The previous code is wrong. `output[i]` returns `np.ndarray` instead of a single  element.

Reply:
I recommend to use [numpy.vectorize](https://numpy.org/doc/stable/reference/generated/numpy.vectorize.html) to wrap the function which are applied to a numpy array elementwise.

Reply:
Great! Done.